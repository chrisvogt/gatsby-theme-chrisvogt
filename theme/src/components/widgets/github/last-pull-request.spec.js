import React from 'react'
import renderer from 'react-test-renderer'
import LastPullRequest from './last-pull-request'

describe('Widget/GitHub/LastPullRequest', () => {
  describe('snapshots', () => {
    it('matches the loading state snapshot', () => {
      const tree = renderer
        .create(<LastPullRequest pullRequest={{}} isLoading />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('matches the repository variant snapshot', () => {
      const mockPullRequest = {
        number: 42,
        repository: {
          name: 'Fake Project'
        },
        title: "Add fake information to the fake project's repository",
        url: 'https://www.github.com/chrisvogt/null/pulls/42'
      }
      const tree = renderer
        .create(<LastPullRequest pullRequest={mockPullRequest} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
