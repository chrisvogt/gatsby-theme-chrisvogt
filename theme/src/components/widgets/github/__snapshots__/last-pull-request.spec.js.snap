// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget/GitHub/LastPullRequest snapshots matches the loading state snapshot 1`] = `
<div
  className="css-vurnku"
>
  <h3
    className="css-4bdn9t-LastPullRequest"
  >
    Last Pull Request
  </h3>
  <a
    className="css-1u3g0h0-LastPullRequest"
  >
    <div
      className="css-1wbxd3x-LastPullRequest"
    >
      <div
        className="text-row show-loading-animation"
        style={
          Object {
            "backgroundColor": "#efefef",
            "height": "1em",
            "marginTop": 0,
            "maxHeight": undefined,
            "width": "100%",
          }
        }
      />
      <div
        className="css-1nldqfs-CardFooter"
      >
        <div
          className="text-row show-loading-animation"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "1em",
              "marginTop": 0,
              "maxHeight": undefined,
              "width": "140px",
            }
          }
        />
      </div>
    </div>
  </a>
</div>
`;

exports[`Widget/GitHub/LastPullRequest snapshots matches the repository variant snapshot 1`] = `
<div
  className="css-vurnku"
>
  <h3
    className="css-4bdn9t-LastPullRequest"
  >
    Last Pull Request
  </h3>
  <a
    className="css-1u3g0h0-LastPullRequest"
    href="https://www.github.com/chrisvogt/null/pulls/42"
  >
    <div
      className="css-1wbxd3x-LastPullRequest"
    >
      <span>
        Add fake information to the fake project's repository
         (
        <span
          className="css-1qk98nv-LastPullRequest"
        >
          #
          42
        </span>
        ) â€“ in 
        <em>
          Fake Project
        </em>
      </span>
      <div
        className="css-1nldqfs-CardFooter"
      >
        <span>
          View on 
          GitHub
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-external-link-alt fa-w-16 "
            data-icon="external-link-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </a>
</div>
`;
